File: src/main/java/instance/os/InstanceSnapshot.java
package instance.os;

import org.jfree.chart.JFreeChart;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-04-08
 *
 */

public class InstanceSnapshot extends Snapshot {

	private JFreeChart cpuChart;
	private JFreeChart bandwidthChart;
	private String log;
	
	public InstanceSnapshot(int id) {
		super(id);
	}
	
	public void addCPULoadChart(JFreeChart chart) {
		this.cpuChart = chart;
	}
	
	public void addBandwidthChart(JFreeChart chart) {
		this.bandwidthChart = chart;
	}
	
	public void addLogText(String log) {
		this.log = log;
	}

	public JFreeChart getCpuChart() {
		return cpuChart;
	}

	public JFreeChart getBandwidthChart() {
		return bandwidthChart;
	}

	public String getLog() {
		return log;
	}
	
}

File: src/main/java/instance/os/InstanceSnapshot.scala
package instance.os

import org.jfree.chart.JFreeChart

/**
 * @author Amir Moulavi
 * @date 2011-07-16
 *
 */
class InstanceSnapshot(id:Int) extends Snapshot(id) {

  private var cpuChart:JFreeChart = _
  private var bandwidthChart:JFreeChart = _
  private var log:String = _

  def addCPULoadChart(chart: JFreeChart) = this.cpuChart = chart

  def addBandwidthChart(chart: JFreeChart) = this.bandwidthChart = chart

  def addLogText(log: String) = this.log = log

  def getCpuChart: JFreeChart = cpuChart

  def getBandwidthChart: JFreeChart = bandwidthChart

  def getLog: String = log

  def getId = id

}